commands.go

```diff
@@ -1000,16 +1145,159 @@ func fnOpen(ce *WrappedCommandEvent) {
-		if err != nil {
-			ce.Reply("Failed to create room: %v", err)
+		if bridgeRoomID == "" {
+			err = portal.CreateMatrixRoom(ce.User, info, true, true)
+			if err != nil {
+				ce.Reply("Failed to create room: %v", err)
+			} else {
+				ce.Reply("Portal room created.")
+			}
 		} else {
-			ce.Reply("Portal room created.")
+			// TODO Move to a function
+			ce.User.CommandState = map[string]interface{}{
+				"next":         &StateHandler{confirmBridge, "Room bridging"},
+				"bridgeToMXID": bridgeRoomID,
+				"jid":          info.JID,
+			}
+			ce.Reply("That WhatsApp group has no existing portal. To confirm bridging the " +
+				"group, use `$cmdprefix  continue`. To cancel, use `$cmdprefix  cancel`.",
+			)
+		}
+	}
+}
+
+func cleanupOldPortalWhileBridging(ce *WrappedCommandEvent, portal *Portal) (bool, func()) {
+	if len(portal.MXID) == 0 {
+		ce.Reply("The portal seems to have lost its Matrix room between you" +
+			"calling `$cmdprefix  bridge` and this command.\n\n" +
+			"Continuing without touching the previous Matrix room...",
+		)
+		return true, nil
+	}
+	switch ce.Args[0] {
+	case "delete-and-continue":
+		return true, func() {
+			portal.Cleanup("Portal deleted (moving to another room)", false)
+		}
+	case "unbridge-and-continue":
+		return true, func() {
+			portal.Cleanup("Room unbridged (portal moving to another room)", true)
+		}
+	default:
+		ce.Reply("The chat you were trying to bridge already has a Matrix portal room.\n\n" +
+			"Please use `$cmdprefix  delete-and-continue` or `$cmdprefix  unbridge-and-" +
+			"continue` to either delete or unbridge the existing room (respectively) and " +
+			"continue with the bridging.\n\n" +
+			"If you changed your mind, use `$cmdprefix  cancel` to cancel.",
+		)
+		return false, nil
+	}
+}
+
+func confirmBridge(ce *WrappedCommandEvent) {
+	defer func() {
+		if err := recover(); err != nil {
+			ce.User.CommandState = nil
+			ce.Reply("Fatal error: %v. This shouldn't happen unless you're messing with the command handler code.", err)
+		}
+	}()
+
+	status := ce.User.GetCommandState()
+	bridgeToMXID := status["bridgeToMXID"].(id.RoomID)
+	portal := ce.User.GetPortalByJID(status["jid"].(types.JID))
+	if portal == nil {
+		panic("could not retrieve portal that was expected to exist")
+	}
+
+	_, mxidInStatus := status["mxid"]
+	if mxidInStatus {
+		ok, f := cleanupOldPortalWhileBridging(ce, portal)
+		if !ok {
+			return
+		} else if f != nil {
+			go f()
+			ce.Reply("Cleaning up previous portal room...")
 		}
+	} else if len(portal.MXID) > 0 {
+		ce.User.CommandState = nil
+		ce.Reply("The portal seems to have created a Matrix room between you " +
+			"calling `$cmdprefix  bridge` and this command.\n\n" +
+			"Please start over by calling the bridge command again.",
+		)
+		return
+	} else if ce.Args[0] != "continue" {
+		ce.Reply("Please use `$cmdprefix  continue` to confirm the bridging or " +
+			"`$cmdprefix  cancel` to cancel.",
```

Comment:
Are the double spaces intended?
```suggestion
			"calling `$cmdprefix bridge` and this command.\n\n" +
			"Please start over by calling the bridge command again.",
		)
		return
	} else if ce.Args[0] != "continue" {
		ce.Reply("Please use `$cmdprefix continue` to confirm the bridging or " +
			"`$cmdprefix cancel` to cancel.",
```

Reply:
Yes, because of a bug in `mautrix-go`.  Without the spaces, the message is printed as `!wacancel`.  I'll submit a fix there when I get a chance.

`$cmdprefix+sp` isn't supported either, sadly.